rules/code-quality/tests/Rector/FuncCall/ArrayKeysAndInArrayToArrayKeyExistsRector/AbstractRunnableTestCase.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\Tests\Rector\FuncCall\ArrayKeysAndInArrayToArrayKeyExistsRector;
+
+use Nette\Utils\Strings;
+use Rector\Core\Testing\PHPUnit\AbstractRectorTestCase;
+use Rector\Core\Testing\ValueObject\SplitLine;
+use ReflectionClass;
+use Symplify\SmartFileSystem\SmartFileInfo;
+
+abstract class AbstractRunnableTestCase extends AbstractRectorTestCase
+{
+    protected function assertOriginalAndFixedFileResultEquals(string $file): void
+    {
+        /**
+         * Todo: Duplicate from
+         *
+         * @see FixtureSplitter::splitContentToOriginalFileAndExpectedFile
+         * ==> refactor in a method
+         */
+        $smartFileInfo = new SmartFileInfo($file);
+        if (Strings::match($smartFileInfo->getContents(), SplitLine::SPLIT_LINE)) {
+            [$originalContent, $expectedContent] = Strings::split($smartFileInfo->getContents(), SplitLine::SPLIT_LINE);
+        } else {
+            $originalContent = $smartFileInfo->getContents();
+            $expectedContent = $originalContent;
+        }
+
+        $originalInstance = $this->loadClass($originalContent);
+        if ($originalInstance !== null) {
+            $expectedInstance = $this->loadClass($expectedContent);
+            if ($expectedInstance !== null) {
+                $actual = $originalInstance->run();
+                $expected = $expectedInstance->run();
+
+                $this->assertSame($actual, $expected);
+            }
+        }
+    }
+
+    protected function getTemporaryClassName(): string
+    {
+        $testName = (new ReflectionClass(static::class))->getShortName();
+        // Todo - pull in Ramsey UUID to generate temporay class names?
+//        $uuid      = Uuid::uuid4()->toString();
+        $uuid = md5((string) random_int(0, 100000000));
+        $className = $testName . '_' . $uuid;
+
+        return (string) Strings::replace($className, '#[^0-9a-zA-Z]#', '_');
+    }
+
+    protected function loadClass(string $classContent): ?RunnableInterface
+    {
+        $className = $this->getTemporaryClassName();
+        $loadable = (string) Strings::replace($classContent, '#\\s*<\\?php#', '');
+        $loadable = (string) Strings::replace($loadable, '#\\s*namespace.*;#', '');
+        $loadable = (string) Strings::replace($loadable, '#class\\s+(\\S*)\\s+#', sprintf('class %s ', $className));
+        eval($loadable);
+        if (is_a($className, RunnableInterface::class, true)) {
```

Comment:
This should be right under `$className = $this->getTemporaryClassName();` with early reaturn,
as it can skipp all the `Strings::replace()`

Reply:
Don't think that would work. 
`is_a` will only be able to recognize the class after `eval` has made it available to php

Reply:
Ah, I didn't notice. Is there a way to handle this withou `eval()`?
That opens big black door.

Reply:
Well, to be fair you already have an unchecked `require_once` in `\Rector\Core\Testing\PHPUnit\FixtureSplitter::splitContentToOriginalFileAndExpectedFile` (which is basically the same) :)

Only way to work around this is to split the "orignal" file and the "fixed" files into separate files and also use different class names and proper autoloading. However, this clashes with the current way fixtures are written and checked (e.g. it won't be possible to do a full "equality" check of the contents any longer because the class names will be different - which I think is very helpful).

If you keep the "goal" in mind (i.e. "reuse existing test files and simply add an additional layer of validation), I'm afraid there is no way around using something that dymically injects code.